inference-engine/src/gna_plugin/transformations/convert_padded2valid_conv.cpp

```diff
@@ -276,184 +257,81 @@ ConvertPadded2ValidConv::ConvertPadded2ValidConv() {
-            pattern_map.at(trailing_transpose).get_node_shared_ptr(), pattern_map.at(bias).get_node_shared_ptr(), nullptr, nullptr);
-    };
-
-    auto m = std::make_shared<ngraph::pattern::Matcher>(bias, matcher_name);
-    this->register_matcher(m, callback);
-}
-
-ConvertPaddedTransposedWithBiasAF2ValidConv::ConvertPaddedTransposedWithBiasAF2ValidConv() {
-    MATCHER_SCOPE(ConvertPaddedTransposedWithBiasAF2ValidConv);
-
-    auto const_input = ngraph::pattern::wrap_type<ngraph::opset7::Constant>();
-    auto leading_transpose = ngraph::pattern::wrap_type<ngraph::opset7::Transpose>({ ngraph::pattern::any_input(), const_input },
-        consumers_and_rank(1, 4));
-    auto conv = ngraph::pattern::wrap_type<ngraph::opset7::Convolution>(
-        { leading_transpose, ngraph::pattern::wrap_type<ngraph::opset7::Constant>(ngraph::pattern::rank_equals(4)) },
+    auto fq = ngraph::pattern::wrap_type<ngraph::opset7::FakeQuantize>({bias, const_input, const_input, const_input, const_input},
         ngraph::pattern::consumers_count(1));
-    auto trailing_transpose = ngraph::pattern::wrap_type<ngraph::opset7::Transpose>({ conv, const_input },
-        consumers_and_rank(1, 4));
-    auto bias = ngraph::pattern::wrap_type<ngraph::opset7::Add>({ trailing_transpose, const_input },
-        ngraph::pattern::consumers_count(1));
-    auto af = ngraph::pattern::wrap_type<ngraph::opset7::Relu, ngraph::opset7::Sigmoid,
+    auto af1 = ngraph::pattern::wrap_type<ngraph::opset7::Relu, ngraph::opset7::Sigmoid,
         ngraph::opset7::Tanh, ngraph::opset7::Abs, ngraph::opset7::Log, ngraph::opset7::Exp,
-        ngraph::opset7::Sign, ngraph::opset7::Clamp>({ bias },
-        ngraph::pattern::consumers_count(1));
+        ngraph::opset7::Sign, ngraph::opset7::Clamp>({bias},
+            ngraph::pattern::consumers_count(1));
+    auto af2 = ngraph::pattern::wrap_type<ngraph::opset7::Relu, ngraph::opset7::Sigmoid,
+        ngraph::opset7::Tanh, ngraph::opset7::Abs, ngraph::opset7::Log, ngraph::opset7::Exp,
+        ngraph::opset7::Sign, ngraph::opset7::Clamp>({fq},
+            ngraph::pattern::consumers_count(1));
+    auto root = std::make_shared<ngraph::pattern::op::Or>(ngraph::OutputVector{bias, fq, af1, af2});
 
     ngraph::matcher_pass_callback callback = [=](ngraph::pattern::Matcher& m) {
         const auto& pattern_map = m.get_pattern_value_map();
         return Convert(pattern_map.at(leading_transpose).get_node_shared_ptr(), pattern_map.at(conv).get_node_shared_ptr(),
-            pattern_map.at(trailing_transpose).get_node_shared_ptr(), pattern_map.at(bias).get_node_shared_ptr(),
-            pattern_map.at(af).get_node_shared_ptr(), nullptr);
+            pattern_map.at(trailing_transpose).get_node_shared_ptr(), pattern_map.at(bias).get_node_shared_ptr());
     };
 
-    auto m = std::make_shared<ngraph::pattern::Matcher>(af, matcher_name);
+    auto m = std::make_shared<ngraph::pattern::Matcher>(bias, matcher_name);
```

Comment:
why do we need lines 217-327 if bias is passes to matcher as a root node?

Reply:
We are detecting all cases described in the header file.

Reply:
But these cases will be special cases of the subgraph with bias at the end, are nodes after the bias checked/modified some way in this matcher?

Reply:
You're right. For padding support it may bring not much value. I believe the original author intended it for 2D convolution factorization, but left it here as well. If it's OK I'd leave it here to have broader test coverage available and remove it from here with 2D convolution factorization merge.

Reply:
Actually when Or operation is a root node in the matcher it doesn't work properly since the shortest graph is always selected, so these nodes (after bias) seem not to be included in the subgraph anyway. If they are required to be included in the subgraph separate matchers should be created for all variants in Or.

Reply:
I'll remove the code from here as it needs to be modified anyway.